src/test/java/api/loans/anonymization/AnonymizeLoansImmediatelyAPITests.java

```diff
@@ -0,0 +1,224 @@
+  }
+
+  /**
+   * Scenario 3
+   *
+   *     Given:
+   *         An Anonymize closed loans setting of "Immediately after loan closes"
+   *         An Anonymize closed loans with associated fees/fines setting of "Never"
+   *         An open loan with an associated fee/fine
+   *     When the item in the loan is checked in
+   *     Then do not anonymize the loan
+   */
+  @Test
+  public void testClosedLoanWithOpenFeesAndFinesIsNotAnonymizedForSettingsOfNever()
+    throws InterruptedException, ExecutionException, TimeoutException, MalformedURLException {
+
+    LoanHistoryConfigurationBuilder loanHistoryConfig = new LoanHistoryConfigurationBuilder()
+      .loanCloseAnonymizeImmediately()
+      .feeFineCloseAnonymizeNever();
+    createConfiguration(loanHistoryConfig);
+
+    IndividualResource loanResource = loansFixture.checkOutByBarcode(new CheckOutByBarcodeRequestBuilder().forItem(item1)
+      .to(user)
+      .at(servicePoint.getId()));
+    UUID loanID = loanResource.getId();
+
+    createOpenAccountWithFeeFines(loanResource);
+
+    loansFixture.checkInByBarcode(item1);
+
+    anonymizeLoansInTenant();
+
+    assertThat(loansStorageClient.getById(loanID).getJson(),
+      not(isAnonymized()));
+  }
+
+  /**
+   * Scenario 4
+   *
+   *     Given:
+   *         An Anonymize closed loans setting of "Immediately after loan closes"
+   *         An Anonymize closed loans with associated fees/fines setting of "Never"
+   *         An open loan with an associated fee/fine
+   *     When all fees/fines associated with the loan are closed
+   *     Then do not anonymize the loan
+   */
+  @Test
+  public void testOpenLoanWithClosedFeesAndFinesIsNotAnonymizedForSettingsOfNever()
+    throws InterruptedException, ExecutionException, TimeoutException, MalformedURLException {
+
+    LoanHistoryConfigurationBuilder loanHistoryConfig = new LoanHistoryConfigurationBuilder()
+      .loanCloseAnonymizeImmediately()
+      .feeFineCloseAnonymizeNever();
+    createConfiguration(loanHistoryConfig);
+
+    IndividualResource loanResource = loansFixture.checkOutByBarcode(new CheckOutByBarcodeRequestBuilder().forItem(item1)
+      .to(user)
+      .at(servicePoint.getId()));
+    UUID loanID = loanResource.getId();
+
+    createClosedAccountWithFeeFines(loanResource, DateTime.now());
+
+    loansFixture.checkInByBarcode(item1);
+
+    anonymizeLoansInTenant();
+
+    assertThat(loansStorageClient.getById(loanID).getJson(),
+      not(isAnonymized()));
+  }
+
+  /**
+   * Scenario 1.1
+   *
+   *     Given:
+   *         An Anonymize closed loans setting of "Immediately"
+   *         An Anonymize closed loans with associated fees/fines setting of "X interval after fee/fine closes"
+   *         An open loan with an associated fee/fine
+   *     When the item in the loan is checked in
+   *     Then do not anonymize the loan
+   */
+  @Test
+  public void testClosedLoanWithClosedFeesAndFinesIsNotAnonymizedWhenIntervalAfterFeeFineClosesNotPassed()
```

Comment:
public void shouldNotAnonymizeOPENLoanWithOpenFeeFineAndAnonymizationIsDeferred()

Reply:
When the scenario contains a phrase "an open loan", but then there also a statement "the loan is checked in" - that is the closed loan, which is reflected in the test name.

Reply:
Test names have been updated